backends/bmv2/run-bmv2-test.py

```diff
@@ -220,13 +235,13 @@ def process_file(options, argv):
         raise Exception("No such file " + options.p4filename)
 
     if options.usePsa:
-        binary = "./p4c-bm2-psa"
+        binary = options.compilerSrcDir + "/build/p4c-bm2-psa"
     else:
-        binary = "./p4c-bm2-ss"
+        binary = options.compilerSrcDir + "/build/p4c-bm2-ss"
```

Comment:
There should also be a way to specify the compiler build subdirectory, which might not be `build` -- I often have `build-release` and `build-debug` directories.  This will prevent running tests at all in such directories.

Reply:
I think we even can replace compilerSrcDir with compilerBuildDir entirely with a couple minor changes. Is that okay? How is `options.testName`, from a superficial look it seems to specify the output test folder?

Reply:
That would seem to be fine.  `options.testName` is only set if the name on the command line includes the path to the compiler source directory (so depends on having that), implying that you might be able to use the script to run tests that aren't in the compiler tree?  Probably for people trying to make new tests or get them from the tutorials or some such.

Not sure how many different ways people regularly use the test scripts -- everyone ends up having their own preferred workflow.

Reply:
Yes, it is a little bit inconvenient to run new tests because they require the python file to be executed from the build folder. I would also like to write external tools that call `run-bmv2-test.py`.
Hmm the downside of switching to `compilerBuildDir`is that the command that generates the `.test` files in `P4CUtils.cmake` has to be updated. I guess that is a larger change impacting not only the bmv2 backend. So we are probably better off leaving this as is. I can add the option.